Modules/cjkcodecs/_codecs_cn.c

```diff
@@ -51,6 +51,8 @@
         ;                                                              \
     }
 
+#define STATE_OFFSET 0
+
 /*
```

Comment:
I felt `c[STATE_OFFSET]` was more explicit than `c[0]`, but don't mind inlining it if you prefer. Alternatively, `STATE_POSITION` may be a better name, as I wanted to get across that only byte 0 is relevant in `state->c`.

Reply:
OK, I get it: `c[8]` is the merged version of the previous union, so it serves two purposes: it's a character array and a (very) small integer. IMHO this (and how it's used) should be documented where `MultibyteCodec_State` is defined.

Reply:
If I were to document how I see `MultibyteCodec_State` now, I would say something along the lines of:
```
A union that provides 8 bytes of state for multibyte codecs. Codecs
are free to use this how they want. Note: if you need to add a new
member to this union, ensure that its byte order is independent of CPU
endianness so that the return value of `getstate` doesn't differ
between little and big endian CPUs.
```

Should I add this as a comment next to the union?

Reply:
Sounds good to me. However `MultibyteCodec_State` no longer is a union, "if you need to add a new member to this union" should be phrased differently.

And `STATE_OFFSET` would have a comment like: "Codecs that have to store a small integer in `MultibyteCodec_State` store it at the following offset".

And yes, `multibytecodec.h` is a little light on documentation, but that's probably another patch.

Reply:
I've added comments next to `STATE_OFFSET` (now `CN_STATE_OFFSET` for clarity) and `MultibyteCodec_State`. Please let me know if the explanations make sense.

Reply:
Looks good to me, but I'd like @methane to make the final decision about merging this PR.